### ISSUE TITLE
Can't find Dependencies when using named parameters in constructor

### ISSUE THREAD

**Issue by GabrielCR99** [Flutterando/auto_injector#6]
**Describe the bug**
When we use a constructor with the new modular injector, e.g 
```ProductsControllerBase({required ProductsRepository productsRepository}) : _productsRepository = productsRepository;```
it can't find the required ProductsRepository dependency.
BUT if we use a non-named parameter, e.g ```ProductsControllerBase(this._productsRepository);``` it works.
Wasn't it supposed to work with both of the constructors?

**Environment**
Add your `flutter doctor -v`

```
[✓] Flutter (Channel stable, 3.10.6, on Microsoft Windows [Version 10.0.22621.2070], locale en-US)
    • Flutter version 3.10.6 on channel stable at C:\src\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f468f3366c (3 weeks ago), 2023-07-12 15:19:05 -0700
    • Engine revision cdbeda788a
    • Dart version 3.0.6
    • DevTools version 2.23.1

[✓] Windows Version (Installed version of Windows is version 10 or higher)

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at C:\Users\gabri\AppData\Local\Android\sdk
    • Platform android-33, build-tools 33.0.0
    • Java binary at:
      C:\Users\gabri\AppData\Local\JetBrains\Toolbox\apps\AndroidStudio\ch-0\223.8836.35.2231.10406996\jbr\bin\java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-b2043.56-10027231)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[✓] Visual Studio - develop for Windows (Visual Studio Community 2022 17.6.5)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.6.33829.357
    • Windows 10 SDK version 10.0.22621.0

[✓] Android Studio (version 2022.2)
    • Android Studio at C:\Users\gabri\AppData\Local\JetBrains\Toolbox\apps\AndroidStudio\ch-0\222.4459.24.2221.10121639
    • Flutter plugin version 74.0.2
    • Dart plugin version 222.4582
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-b2043.56-9586694)

[✓] Android Studio (version 2022.3)
    • Android Studio at C:\Users\gabri\AppData\Local\JetBrains\Toolbox\apps\AndroidStudio\ch-0\223.8836.35.2231.10406996
    • Flutter plugin version 74.0.3
    • Dart plugin version 223.8950
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-b2043.56-10027231)

[✓] IntelliJ IDEA Ultimate Edition (version 2023.2)
    • IntelliJ at C:\Users\gabri\AppData\Local\JetBrains\Toolbox\apps\IDEA-U\ch-0\232.8660.185
    • Flutter plugin version 74.0.5
    • Dart plugin version 232.8660.129

[✓] VS Code (version 1.81.0)
    • VS Code at C:\Users\gabri\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.70.0

[✓] Connected device (4 available)
    • sdk gphone64 x86 64 (mobile) • emulator-5554 • android-x64    • Android 13 (API 33) (emulator)
    • Windows (desktop)            • windows       • windows-x64    • Microsoft Windows [Version 10.0.22621.2070]
    • Chrome (web)                 • chrome        • web-javascript • Google Chrome 115.0.5790.111
    • Edge (web)                   • edge          • web-javascript • Microsoft Edge 115.0.1901.188

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

Add your `pubspec.yaml`
```
name: delivery_backoffice_dw10
description: A new Flutter project.

publish_to: "none"

version: 1.0.0+1

environment:
  sdk: ^3.0.6

dependencies:
  awesome_snackbar_content: ^0.1.3
  brasil_fields: ^1.13.1
  dio: ^5.3.1
  flutter:
    sdk: flutter
  flutter_dotenv: ^5.1.0
  flutter_mobx: ^2.0.6+5
  flutter_modular: ^6.0.1
  flutter_web_plugins:
    sdk: flutter
  loading_animation_widget: ^1.2.0+4
  mobx: ^2.2.0
  url_strategy: ^0.2.0
  validatorless: ^1.2.3

dev_dependencies:
  build_runner: ^2.4.6
  dart_code_metrics: ^5.7.6
  flutter_test:
    sdk: flutter
  mobx_codegen: ^2.3.0

flutter:
  uses-material-design: true

  assets:
    - .env
    - assets/images/
    - assets/images/icons/

  fonts:
    - family: mplus1
      fonts:
        - asset: assets/fonts/MPLUS1p-Light.ttf
          weight: 300
        - asset: assets/fonts/MPLUS1p-Regular.ttf
          style: normal
        - asset: assets/fonts/MPLUS1p-Medium.ttf
          weight: 500
        - asset: assets/fonts/MPLUS1p-Bold.ttf
          weight: 700
        - asset: assets/fonts/MPLUS1p-ExtraBold.ttf
          weight: 800
```
**To Reproduce**
```ProductsControllerBase({required ProductsRepository productsRepository}) : _productsRepository = productsRepository;``` -> Doesn't work
 ```ProductsControllerBase(this._productsRepository);```  -> Works

**Expected behavior**
Should work with both constructors

**Screenshots**
![image](https://github.com/Flutterando/modular/assets/42328188/97c7ac02-3193-427b-92b5-09d190ebd2be)
![image](https://github.com/Flutterando/modular/assets/42328188/793246eb-0f73-437e-9a45-435c4a6af3eb)

**Comment by davidsdearaujo** [comment]
Hi @GabrielCR99 Thanks for reporting this bug!
A new way to read the parameters is being implemented on the PR #6.
Let's see if it's gonna fix your issue, otherwise I'll open a new PR to fix it!

**Comment by davidsdearaujo** [comment]
It was fixed in the last deployed version.

Feel free to open the issue again if you still have the reported problem!
